ui/app/src/state/actions/system.ts

```diff
@@ -126,5 +126,10 @@ export const fetchGlobalMenuFailed = /*#__PURE__*/ createAction('FETCH_GLOBAL_ME
 export const blockUI = /*#__PURE__*/ createAction<Partial<UIBlockerStateProps>>('BLOCK_UI');
 export const unblockUI = /*#__PURE__*/ createAction('UNBLOCK_UI');
 
-export const openSiteSocket = /*#__PURE__*/ createAction<{ site: string; xsrfToken: string }>('OPEN_SITE_SOCKET');
+export const openSiteSocket = /*#__PURE__*/ createAction<{ site?: string; xsrfToken: string }>('OPEN_SITE_SOCKET');
 export const setSiteSocketStatus = /*#__PURE__*/ createAction<{ connected: boolean }>('SET_SITE_SOCKET_STATUS');
+
+// region sites events
+export const siteReady = /*#__PURE__*/ createAction<SocketRootEventBase & { siteId: string }>('SITE_READY_EVENT');
+export const siteDeleting = /*#__PURE__*/ createAction<SocketRootEventBase & { siteId: string }>('SITE_DELETING_EVENT');
```

Comment:
```suggestion
export const newProjectReady = /*#__PURE__*/ createAction<SocketRootEventBase & { siteId: string }>('NEW_PROJECT_READY');
export const projectBeingDeleted = /*#__PURE__*/ createAction<SocketRootEventBase & { siteId: string }>('PROJECT_BEING_DELETED');
```

Reply:
I can't change the types (`SITE_READY_EVENT`, `SITE_DELETING_EVENT`) since those are the event types coming from the websocket

Reply:
(You could have. The event name on the UI doesn't have to match the backend's after it's published internally since the UI doesn't work directly off the backend event but rather from what the worker publishes. But ok, leave it.)